src/circuit/tx.rs

```diff
@@ -175,14 +187,52 @@ pub fn c_transfer<C:CS, P:PoolParams<Fr=C::Fr>>(
         (&s.tx.input.1[i].p_d - c_derive_key_p_d(&s.tx.input.1[i].d.as_num(), &eta_bits, params).x).assert_zero();
     }
 
+    // Check daily turnover limit and transfer limit
+    let in_account = &s.tx.input.0;
+    let out_account = &s.tx.output.0;
+    {
+        let is_new_day = c_comp(&p.day.as_num(), &in_account.last_action_day.as_num(), DAY_SIZE_BITS);
+
+        let delta_value_is_positive = c_comp(&total_value, &p.derive_const(&Num::ZERO), TURNOVER_SIZE_BITS);
+        let delta_value_abs = total_value.clone().switch(&delta_value_is_positive, &-&total_value);
+        let tx_turnover = delta_value_abs + &out_notes_sum;
+        let turnover = tx_turnover.switch(&is_new_day, &(in_account.daily_turnover.as_num() + &tx_turnover));
+
+        // Check transfer limit
+        c_comp(&out_notes_sum, &p.transfer_limit.as_num(), BALANCE_SIZE_BITS).assert_const(&false);
+        // Check that current day >= last_action_day
+        (&is_new_day | p.day.is_eq(&in_account.last_action_day)).assert_const(&true);
+        // Check turnover limit
+        c_comp(&turnover, &p.daily_limit.as_num(), TURNOVER_SIZE_BITS).assert_const(&false);    
+        // Check output account turnover
+        out_account.daily_turnover.as_num().is_eq(&turnover).assert_const(&true);
+        // Check output account last_action_day
+        out_account.last_action_day.as_num().is_eq(&p.day.as_num()).assert_const(&true);
+    }
+
+    //assuming input_pos_index <= current_index
+    let ref input_pos_index = c_from_bits_le(s.in_proof.0.path.as_slice());
+
+    // Check in_account_hash. If account is old, last_action_day and daily_turnover must be zero
+    let (in_account_hash, nullifier) = {
+        let in_account_hash_new = in_account.hash(params);
+        let in_account_hash_old = in_account.hash_old(params);
+
+        let nullifier_new = c_nullfifier(&in_account_hash_new, &eta, input_pos_index, params);
+        let nullifier_old = c_nullfifier(&in_account_hash_old, &eta, input_pos_index, params);

```

Comment:
Is it more optimal to calculate two nullifiers and then choose the right one than choosing the right account hash and then calculating the nullifier from it?

Reply:
Could you elaborate? We use a nullifier exactly for choosing the right account hash. It seems that it is the simplest way.  Btw there is no "true" branching in snarks. I mean you always calculate all branches even if only one is needed.

Reply:
My bad. Haven't noticed that `in_account_hash` calculation depends on both nullifiers.